lib/preview.js

```diff
@@ -0,0 +1,252 @@
+    triggerNotify(notify, 'Removed', filePath, deletePromise);
+  };
+};
+
+const buildUploadFileToPreviewBufferCallback = (sessionInfo, notifyMessage) => {
+  const { portalId, src, dest, notify } = sessionInfo;
+
+  return async filePath => {
+    if (!isAllowedExtension(filePath)) {
+      logger.debug(`Skipping ${filePath} due to unsupported extension`);
+      return;
+    }
+    if (shouldIgnoreFile(filePath)) {
+      logger.debug(`Skipping ${filePath} due to an ignore rule`);
+      return;
+    }
+    const destPath = getDesignManagerPath(src, dest, filePath);
+    const uploadPromise = uploadFile(portalId, filePath, destPath);
+    triggerNotify(notify, notifyMessage, filePath, uploadPromise);
+  };
+};
+
+const initialPreviewBufferUpload = async (sessionInfo, filePaths) => {
+  const { portalId, src, dest } = sessionInfo;
+
+  return uploadFolder(portalId, src, dest, fileMapperArgs, {}, filePaths);
+};
+
+const startPreviewWatcher = async sessionInfo => {
+  const { src } = sessionInfo;
+  let watcherIsReady = false;
+
+  const watcher = chokidar.watch(src, {
+    ignoreInitial: true, // makes initial addition of files not trigger the watcher
+    ignored: file => shouldIgnoreFile(file),
+  });
+
+  const addFileCallback = buildUploadFileToPreviewBufferCallback(
+    sessionInfo,
+    'Added'
+  );
+  const changeFileCallback = buildUploadFileToPreviewBufferCallback(
+    sessionInfo,
+    'Change'
+  );
+  const deleteFileCallback = buildDeleteFileFromPreviewBufferCallback(
+    sessionInfo,
+    'file'
+  );
+  const deleteFolderCallback = buildDeleteFileFromPreviewBufferCallback(
+    sessionInfo,
+    'folder'
+  );
+
+  watcher.on('ready', () => {
+    console.log('Local file watching service has started!');
+    watcherIsReady = true;
+  });
+  watcher.on('add', addFileCallback);
+  watcher.on('change', changeFileCallback);
+  watcher.on('unlink', deleteFileCallback);
+  watcher.on('unlinkDir', deleteFolderCallback);
+
+  function sleep(ms) {
+    return new Promise(resolve => {
+      setTimeout(resolve, ms);
+    });
+  }
+  while (!watcherIsReady) {
+    await sleep(1);
+    // do nothing... surely there's a better way to do this...
+  }
+  return watcher;
+};
+
+const createLocalHttpServer = async sessionInfo => {
+  const expressServer = express();
+  //expressServer.use(bodyParser.json());
+  expressServer.use('/', await createPreviewServerRoutes(sessionInfo));
+
+  return expressServer;
+};
+
+const preview = async (
+  accountId,
+  src,
+  dest,
+  { notify, filePaths, skipUpload, noSsl, port }
+) => {
+  const accountConfig = getAccountConfig(accountId);
+  const domains = await getPortalDomains(accountId);
+  const sessionToken = '96cd331a-189d-41f2-8a4c-a12485402eff';
+  const PORT = port || 3000;
+  const protocol = noSsl ? 'http' : 'https';
+
+  const sessionInfo = {
+    src,
+    dest: `@preview/${sessionToken}/${dest}`,
+    portalName: accountConfig.name,
+    portalId: accountId,
```

Comment:
Nit (not a blocker), but we're trying to use `accountId` rather than portalId in local-dev-lib. Just a note for when you port.